src/vtp/ops/run_mock_election_operation.py

```diff
@@ -0,0 +1,474 @@
+                Shellout.run(
+                    ["git", "pull"],
+                    self.parsed_args.printonly,
+                    self.parsed_args.verbosity,
+                    no_touch_stds=True,
+                    timeout=None,
+                    check=True,
+                )
+            if self.parsed_args.flush_mode == 2:
+                merge_contests = MergeContestsOperation(
+                    Shellout.handle_printonly_switch(
+                        [
+                            "-r",
+                            "-f",
+                            "-v",
+                            self.parsed_args.verbosity,
+                        ],
+                        self.parsed_args.printonly,
+                    )
+                )
+                merge_contests.main(the_election_config)
+                tally_contests = TallyContestsOperation(
+                    Shellout.handle_printonly_switch(
+                        [
+                            "-v",
+                            self.parsed_args.verbosity,
+                        ],
+                        self.parsed_args.printonly,
+                    )
+                )
+                tally_contests.main(the_election_config)
+                return
+            merge_contests = MergeContestsOperation(
+                Shellout.handle_printonly_switch(
+                    [
+                        "-r",
+                        "-m",
+                        self.parsed_args.minimum_cast_cache,
+                        "-v",
+                        self.parsed_args.verbosity,
+                    ],
+                    self.parsed_args.printonly,
+                )
+            )
+            merge_contests.main(the_election_config)
+            logging.info("Sleeping for 10")
+            time.sleep(10)
+            elapsed_time = time.time() - start_time
+            if elapsed_time > seconds:
+                break
+        if self.parsed_args.flush_mode in [1, 2]:
+            print("Cleaning up remaining unmerged ballots")
+            merge_contests = MergeContestsOperation(
+                Shellout.handle_printonly_switch(
+                    [
+                        "-r",
+                        "-f",
+                        "-v",
+                        self.parsed_args.verbosity,
+                    ],
+                    self.parsed_args.printonly,
+                )
+            )
+            merge_contests.main(the_election_config)
+        # tally the contests
+        tally_contests = TallyContestsOperation(
+            Shellout.handle_printonly_switch(
+                [
+                    "-v",
+                    self.parsed_args.verbosity,
+                ],
+                self.parsed_args.printonly,
+            )
+        )
+        tally_contests.main(the_election_config)
+
+    ################
+    # main
+    ################
+    # pylint: disable=duplicate-code
+    def main(self, the_election_config):
```

Comment:
This is related to the two currently unresolved conversations above.  Note - it was always a goal not to shell out to other python programs - I just did not prioritize in/for the original demo and was also, until perhaps now, not familiar with the possible design idioms and tradeoffs of doing that.

Regardless, yes, in working on this this week, it was important to pass the 'args' into the constructor so that object itself had them - is saved explicitly passing them around and made the code much easier to move around.  However, after hitting the argparse versus non-argparse conundrum of duplicated code in design space, and backing away from that, I ended up parsing the arguments in a staticmethod.

To me, it was important to have the 'one source of truth' as a staticmethod so that the non-argparse path could use the same exact code as the argparse path.  Before doing that, I was spending time debugging my own code where I had messed up the data that I was passing around in the non argparse case.  If I was messing it, I believe that future generations of consumers, such as VTP-web-api, would mess it up.  And I wanted to phase contain an error at the call and not have it lurk around for some other runtime assert.

So that is the basic reason for where things stand now.